backend/service/audit/sql.go

```diff
@@ -167,6 +170,8 @@ type eventDetails struct {
 	Status            status      `json:"status,omitempty"`
 	RequestResources  []*resource `json:"request_resources,omitempty"`
 	ResponseResources []*resource `json:"response_resources,omitempty"`
+	RequestMetadata   *any.Any    `json:"request_metadata,omitempty"`
+	ResponseMetadata  *any.Any    `json:"response_metadata,omitempty"`
```

Comment:
I initially thought I had to convert this back to a type `interface {}` but then realized that I could keep this as type `*any.Any` when I reviewed the backend/api/audit/v1/audit.pb.go file.
```
# backend/api/audit/v1/audit.pb.go 
type RequestEvent struct {
...
	Resources []*Resource `protobuf:"bytes,6,rep,name=resources,proto3" json:"resources,omitempty"`
	// The API request saved as metadata for the event.
	RequestMetadata *any.Any `protobuf:"bytes,7,opt,name=request_metadata,json=requestMetadata,proto3" json:"request_metadata,omitempty"`
	// The API response saved as metadata for the event.
	ResponseMetadata *any.Any `protobuf:"bytes,8,opt,name=response_metadata,json=responseMetadata,proto3" json:"response_metadata,omitempty"`
}
```

Reply:
unfortunately because proto (and thus `Any`) objects require a special marshaller, you will have to manually marshal them from any first using jsonpb and should not use that type in the database model (which would rely directly on the built-in marshaller).

> [protojson] produces a different output than the standard "encoding/json" package, which does not operate correctly on protocol buffer messages.
source: https://godoc.org/google.golang.org/protobuf/encoding/protojson

from some cursory research, it seems json.RawMessage is commonly used for the database model type JSONB. json.RawMessage is really just a byte array with some helper functions so it can be cast to from the marshaller output, wwhich would look something like
```go
a, _ := ptypes.MarshalAny(pb)
b, _ := protojson.Marshal(a)
r := json.RawMessage(b)
```